File: Easy/1790. Check if One String Swap Can Make Strings Equal/areAlmostEqual.cpp
/*
    time complexity: O(n)
    space complexity: O(n)
*/
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        if(s1 == s2){
            return true;
        }
        set<char> s;
        for(char c : s1){
            s.insert(c);
        }
        int count = 0;
        for(int i = 0; i < s1.size(); i++){
            if(s.count(s2[i]) && s1[i] != s2[i]){
                count++;
            }
        }
        return count == 2? true : false;
    }
};

File: Easy/1790. Check if One String Swap Can Make Strings Equal/areAlmostEqual.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        if(s1.equals(s2)){
            return true;
        }
        HashSet<Character> hs = new HashSet<>();
        for(char c : s1.toCharArray()){
            hs.add(c);
        }
        int count = 0;
        for(int i = 0; i < s1.length(); i++){
            if(hs.contains(s2.charAt(i)) && s1.charAt(i) != s2.charAt(i)){
                count++;
            }
        }
        return count == 2? true : false;
    }
}